tools/integration_tests/read_cache/setup_test.go

```diff
@@ -0,0 +1,97 @@
+package read_cache
```

Comment:
I think this file should name as read_cache_test.go.

Reply:
Let's call it setup_test.go only as this has the setup for the tests. We can rename later if required.